src/Core/Notifications/IDisabledNotificationsStorage.cs

```diff
@@ -32,15 +40,95 @@ public interface IDisabledNotificationsStorage
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class DisabledNotificationsStorage : IDisabledNotificationsStorage
     {
-        public void DisableNotification(string id)
+        private readonly IVsVersionProvider vsVersionProvider;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        private string filePath => GetFilePath();
+
+        private DisabledNotifications disabledNotifications = null;
+        private readonly object lockObject = new object();
+
+        [ImportingConstructor]
+        public DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger) : this(vsVersionProvider, logger, new FileSystem())
         {
-            // todo
+
         }
 
+        internal /*for testing*/ DisabledNotificationsStorage(IVsVersionProvider vsVersionProvider, ILogger logger, IFileSystem fileSystem)
+        {
+            this.vsVersionProvider = vsVersionProvider;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public void DisableNotification(string id)
+        {
+            lock (lockObject)
+            {
+                if (IsNotificationDisabled(id))
+                {
+                    return;
+                }
+                if (disabledNotifications == null)
+                {
+                    return;
+                }
+
+                disabledNotifications.AddNotification(id);
+                SaveNotifications();
+            }
+        }
         public bool IsNotificationDisabled(string id)
         {
-            // todo
-            return false;
+            lock (lockObject)
+            {
+                if (disabledNotifications == null)
+                {
+                    disabledNotifications = ReadDisabledNotifications();
```

Comment:
This is potentially confusing; it will be shown even if the file exists, but couldn't be loaded because of a serialization error.

Reply:
I have changed the wording to "[Notifications]Couldn't load disabled notification config. Notification will be showed."